tensor2tensor/utils/metrics.py

```diff
@@ -285,9 +285,9 @@ def create_evaluation_metrics(problems, model_hparams):
   def make_problem_specific_metric_fn(metric_fn, problem_idx, weights_fn):
     """Create a metric fn conditioned on problem_idx."""
 
-    def problem_metric_fn(predictions, features):
+    def problem_metric_fn(predictions, features, labels):
       """Metric fn."""
-      labels = features.get("targets", None)
+      #labels = features.get("targets", None)
```

Comment:
This will cause problems with current T2T models if I'm not mistaken, is it really needed?

Reply:
This is similar to the change in `_loss_single` - it is not longer true, that labels are always associated with key `targets` - they can be associated with different key. For this reason, we let the caller choose which metric she is interested (this happens in line 1030 and 1032 of t2t_model.py).
Again, I might be missing some context of t2t, so please correct me, if I'm wrong.